### DISCUSSION TITLE
Milvus Build issue

### DISCUSSION THREAD

**Discussion by amazon-aws** [milvus-io/milvus#31043] [Q&A and General discussion]
Hi,

while building milvus , getting below issue as file is not generating conanbuildinfo.cmake

CMake Error at CMakeLists.txt:95 (include):
       include could not find requested file:
     
         /milvus/cmake_build/conan/conanbuildinfo.cmake
     
     
     CMake Error at CMakeLists.txt:97 (conan_basic_setup):
       Unknown CMake command "conan_basic_setup".

Can you suggest on this

**Comment by yhmo** [comment]
This discussion might be helpful: https://stackoverflow.com/questions/60967172/how-to-fix-unknown-command-error-in-cmake-when-i-using-conan

**Reply by amazon-aws** [reply]
@yhmo I have already tried this solution but it didn't work

**Reply by amazon-aws** [reply]
@yhmo Earlier I hadn't removed the previously added generator line. After adding the new line for generator from above link & removing the previous line , it works now. thanks

**Reply by amazon-aws** [reply]
@yhmo Also now I am getting another issue while building milvus: 
/bin/sh: line 1: cargo: command not found
     make[3]: *** [thirdparty/tantivy/CMakeFiles/tantivy_binding_target.dir/build.make:73: thirdparty/tantivy/compile_tantivy] Error 127

although cargo is already installed with version 1.73 but somehow it is giving error "cargo: command not found"

**Reply by yhmo** [reply]
So you are compiling the master branch.
The master branch requires a lib named "tantivy", which is made from rust. So, to compile the master branch, rust and cargo is insrtalled.
I don't know why your cargo is not found. The 3rdparty_build.sh simply install rust and cargo if you don't have: https://github.com/milvus-io/milvus/blob/master/scripts/3rdparty_build.sh#L92

The "$HOME" of my machine is "/home/yhmo". So, when I find the cargo by command "which cargo", it prints the path "/home/yhmo/.cargo/bin/cargo".

You can compile the 2.3 branch instead: https://github.com/milvus-io/milvus/tree/2.3
The 2.3 branch has no tantivy.

**Reply by amazon-aws** [reply]
@yhmo I have compiled using 2.3 branch so above issue is not coming now. Thanks.

I am getting below issue while building milvus using 2.3 codebase.

 go: downloading github.com/ardielle/ardielle-go v1.5.2
     go: updates to go.mod needed; to update it:
     	go mod tidy
     make: *** [Makefile:64: milvus] Error 1

Can you suggest on this ?

**Reply by xiaofan-luan** [reply]
go mod tidy?

**Reply by amazon-aws** [reply]
Sorry, not getting you.

**Reply by yhmo** [reply]
@amazon-aws 
Just run command  "go mod tidy" in the terminal, under the milvus root path.

**Reply by amazon-aws** [reply]
@xiaofan-luan I have added go mod tidy in install_deps.sh script and still getting below error :                                                             Go Version: go version go1.21.8 linux/amd64
     Building Milvus ...
     go: updates to go.mod needed; to update it:
         go mod tidy
     make: *** [Makefile:64: milvus] Error 1 
I am building milvus using 2.3 codebase through automation .

where should I  add command "go mod tidy"   install_deps.sh or MakeFile

**Reply by yhmo** [reply]
Normally, no need to add "do mod tidy" into install_deps.sh or MakeFile. Something wrong with your environment.

Try:
`go clean -modcache`
then:
`go mod tidy`

**Reply by amazon-aws** [reply]
@yhmo  Thanks for the heads up. after successful make execution, I can see bin directory containing milvus. Can you suggest what are steps further to publish image to artifactory

**Reply by yhmo** [reply]
Steps:
1. make   ---------------- I suppose you have successfully run command "make"
2. make install -------------- this command copy the necessary libraries into "lib" directory under the milvus root path
3. ./build/build_image.sh -------------- this script will generate a docker image milvusdb/milvus:latest
![Screenshot from 2024-03-18 11-31-01](https://github.com/milvus-io/milvus/assets/2282099/cae0efe9-53fa-4e63-a672-7c9554dcb2c1)

**Reply by amazon-aws** [reply]
@yhmo  while generating docker image for milvus, getting below  issue 

Step 13/35 : RUN chmod +x /tini
      ---> Running in 84cfdbf1f3f5
     /bin/sh: /lib/x86_64-linux-gnu/libc.so.6: version GLIBC_2.32' not found (required by /milvus/lib/libjemalloc.so)
     /bin/sh: /lib/x86_64-linux-gnu/libc.so.6: version GLIBC_2.34' not found (required by /milvus/lib/libjemalloc.so)
     The command '/bin/sh -c chmod +x /tini' returned a non-zero code: 1
<-- Completed package:dockerfile step in 0:28.5

Can you please suggest on this

**Reply by yhmo** [reply]
I have no idea why you get the error.
This is the readme for how to build milvus by docker:
https://github.com/milvus-io/milvus/blob/2.3/build/README.md

**Comment by xiaofan-luan** [comment]
> Hi,
> 
> while building milvus , getting below issue as file is not generating conanbuildinfo.cmake
> 
> CMake Error at CMakeLists.txt:95 (include): include could not find requested file:
> 
> ```
>      /milvus/cmake_build/conan/conanbuildinfo.cmake
>  
>  
>  CMake Error at CMakeLists.txt:97 (conan_basic_setup):
>    Unknown CMake command "conan_basic_setup".
> ```
> 
> Can you suggest on this

your github id is cool my man

**Reply by xiaofan-luan** [reply]
did you run install_dep.sh before you build?

**Reply by amazon-aws** [reply]
@xiaofan-luan when I am executing docker file after milvus bin get generated and getting below issue as I have tried multiple solution to resolve it. Can you suggest for same 
/bin/sh: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /milvus/lib/libjemalloc.so)

**Reply by xiaofan-luan** [reply]
I think you need to use amazonlinux:2023 or ubuntu 20.04 or above.

What is the linux release you are using? CentOS8 is longer not supported since it is already EOL.

You kind like have to upgrade your glibc

**Reply by amazon-aws** [reply]
@xiaofan-luan After updating the ubuntu version to 22.04, milvus artifact created. Thanks for the heads up.

Can you confirm using which branch "milvus 2.3" branch created ?

**Reply by xiaofan-luan** [reply]
branch 2.3 is the latest dev branch for milvus.
lastest tag is v2.3.13

**Reply by amazon-aws** [reply]
okay

**Reply by amazon-aws** [reply]
@xiaofan-luan : Last week milvus was build successfully uisng 2.3 codebase but yesterday, when execute this codebase again, it is giving error as this URL  https://github.com/tukaani-project/xz/archive/v5.4.4.tar.gz is disabled by github staff and due to which milvus bin& libs are not generating. I have got the new url details : https://sourceforge.net/projects/lzmautils/files but not able to find from which  milvus codebase file/folder calling . Can you suggest on this ?

**Reply by xiaofan-luan** [reply]
simply disable the azure build, there is an build option for that

**Reply by amazon-aws** [reply]
@xiaofan-luan where to disable the azure build

**Reply by xiaofan-luan** [reply]
try make ENABLE_AZURE=false

**Reply by amazon-aws** [reply]
@xiaofan-luan ok but in which file in milvus codebase this param "ENABLE_AZURE" is existing

**Reply by xiaofan-luan** [reply]
core_build.sh

you can grep on the code base, doesn't worth to ask.

**Reply by xiaofan-luan** [reply]
And i would highly recommend to use zilliz cloud if possible, saves you tons of trouble~

**Reply by amazon-aws** [reply]
@xiaofan-luan yes, I have already checked in core_build.sh but there is no such parameter so asked you.
I have found this param  "BUILD_WITHOUT_AZURE="on" " only

**Reply by amazon-aws** [reply]
@xiaofan-luan : Executing  "make ENABLE_AZURE=false" creating issues, pls see below error. Till this repo : https://github.com/tukaani-project/xz/archive/v5.4.4.tar.gz disabled by Github , everything was fine. can you share latest version of  milvus codebase.

-- Running vcpkg install - failed
     CMake Error at /workspace/cmake_build/azure/_deps/vcpkg-src/scripts/buildsystems/vcpkg.cmake:899 (message):
       vcpkg install failed.  See logs for more information:
       /workspace/cmake_build/azure/vcpkg-manifest-install.log
     Call Stack (most recent call first):
       /usr/local/share/cmake-3.28/Modules/CMakeDetermineSystem.cmake:170 (include)
       CMakeLists.txt:22 (project)
     
     
     CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
     CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
     CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
     -- Configuring incomplete, errors occurred!
     make[1]: Entering directory '/workspace/cmake_build/azure'
     make[1]: *** No rule to make target 'install'.  Stop.
     make[1]: Leaving directory '/workspace/cmake_build/azure'
     build blob-chunk-manager fail...
     Downloading vcpkg-glibc...
     vcpkg package management program version 2023-10-18-27de5b69dac4b6fe8259d283cd4011e6d20a84ce

**Reply by xiaofan-luan** [reply]
if you run with "make ENABLE_AZURE=false", you should not see the following vcpkg install thing.
<img width="282" alt="image" src="https://github.com/milvus-io/milvus/assets/83447078/62b0b008-d291-45b4-ba79-693cb937fe71">

**Reply by xiaofan-luan** [reply]
ENABLE AZURE will set -Z tag and BUILD_WITHOUT_AZURE will be set to ON

**Reply by amazon-aws** [reply]
@xiaofan-luan Yes, it shouldn't run the vcpkg install but as I have shared the logs, still it is creating issues. How milvus will build using latest codebase if dependency exists. It will create issue on every platform

**Reply by xiaofan-luan** [reply]
I actually works on my laptop. So I guess you need to check.
Also the vcpkg works fine on linux system.

**Reply by amazon-aws** [reply]
@xiaofan-luan : you mean to say that when you are executing "make ENABLE_AZURE=false" then milvus is building successfully ?

**Reply by xiaofan-luan** [reply]
I my laptop, both "make ENABLE_AZURE=false" or ""make ENABLE_AZURE=true" works.
and make ENABLE_AZURE=false will skip the vcpkg install

**Reply by xiaofan-luan** [reply]
and it's the same all our ci/cd environemnt. So I guess you need to figure out this by yourselves, sorry about that.

**Reply by xiaofan-luan** [reply]
the xz problem is already fixed by changing vcpkg source https://github.com/milvus-io/milvus/pull/31770

**Reply by amazon-aws** [reply]
@xiaofan-luan: they have pushed to master branch so I pull this file "AzureVcpkg.cmake" and push it to my repo & build using the same. Right ?

**Comment by xiaofan-luan** [comment]
do go mod tidy?
